### ISSUE TITLE
[Bug]: dotnet nuget push --help does not show proper help text 

### ISSUE THREAD

**Issue by joelverhagen** [NuGet/Home#11714]
### NuGet Product Used

dotnet.exe

### Product Version

6.0.300-preview.22154.4

### Impact

It's more difficult to complete my work

### Repro Steps & Context

`dotnet nuget push --help`

```
PS C:\z\GitHub\NuGet> dotnet nuget push --help
Description:

Usage:
  dotnet nuget push <ienumerable`1>... [options]

Arguments:
  <ienumerable`1>

Options:
  --force-english-output
  -s, --source <source>
  -ss, --symbol-source <symbol-source>
  -t, --timeout <timeout>
  -k, --api-key <api-key>
  -sk, --symbol-api-key <symbol-api-key>
  -d, --disable-buffering
  -n, --no-symbols
  --no-service-endpoint
  --interactive
  --skip-duplicate
  -?, -h, --help                          Show command line help.
```

Strangely, this works:
```
PS C:\z\GitHub\NuGet> dotnet nuget push --info
Specify --help for a list of available options and commands.
error: Unrecognized option '--info'


Usage: dotnet nuget push [arguments] [options]

Arguments:
  [root]  Specify the path to the package and your API key to push the package to the server.

Options:
  -h|--help                      Show help information
  --force-english-output         Forces the application to run using an invariant, English-based culture.
  -s|--source <source>           Package source (URL, UNC/folder path or package source name) to use. Defaults to DefaultPushSource if specified in NuGet.Config.
  -ss|--symbol-source <source>   Symbol server URL to use.
  -t|--timeout <timeout>         Timeout for pushing to a server in seconds. Defaults to 300 seconds (5 minutes).
  -k|--api-key <apiKey>          The API key for the server.
  -sk|--symbol-api-key <apiKey>  The API key for the symbol server.
  -d|--disable-buffering         Disable buffering when pushing to an HTTP(S) server to decrease memory usage.
  -n|--no-symbols                If a symbols package exists, it will not be pushed to a symbols server.
  --no-service-endpoint          Does not append "api/v2/package" to the source URL.
  --interactive                  Allow the command to block and require manual action for operations like authentication.
  --skip-duplicate               If a package and version already exists, skip it and continue with the next package in the push, if any.
```

**Comment by dominoFire** [comment]
Reproduced in 6.0.300-preview.22178.8

Commmand: `dotnet nuget push --help`

Output:

```
Description:

Usage:
  dotnet nuget push <ienumerable`1>... [options]

Arguments:
  <ienumerable`1>

Options:
  --force-english-output
  -s, --source <source>
  -ss, --symbol-source <symbol-source>
  -t, --timeout <timeout>
  -k, --api-key <api-key>
  -sk, --symbol-api-key <symbol-api-key>
  -d, --disable-buffering
  -n, --no-symbols
  --no-service-endpoint
  --interactive
  --skip-duplicate
  -?, -h, --help                          Show command line help.




```

**Comment by dotnokato** [comment]
Hello,

I was not able to reproduce this issue in `dotnet 7.0.202` with `dotnet nuget version 6.5.0.136`.
Looks like it got fixed somewhere between 6.0.300-preview.22178.8 and 6.5.0.136.

```
❯ dotnet nuget push -h
Description:

Usage:
  dotnet nuget push <package-paths>... [options]

Arguments:
  <package-paths>

Options:
  --force-english-output
  -s, --source <source>
  -ss, --symbol-source <symbol-source>
  -t, --timeout <timeout>
  -k, --api-key <api-key>
  -sk, --symbol-api-key <symbol-api-key>
  -d, --disable-buffering
  -n, --no-symbols
  --no-service-endpoint
  --interactive
  --skip-duplicate
  -?, -h, --help                          Show command line help.
```

```
❯ dotnet nuget push
error: Please specify the path to the package.


Usage: dotnet nuget push [arguments] [options]

Arguments:
  [root]  Specify the path to the package and your API key to push the package to the server.

Options:
  -h|--help                      Show help information
  --force-english-output         Forces the application to run using an invariant, English-based culture.
  -s|--source <source>           Package source (URL, UNC/folder path or package source name) to use. Defaults to DefaultPushSource if specified in NuGet.Config.
  -ss|--symbol-source <source>   Symbol server URL to use.
  -t|--timeout <timeout>         Timeout for pushing to a server in seconds. Defaults to 300 seconds (5 minutes).
  -k|--api-key <apiKey>          The API key for the server.
  -sk|--symbol-api-key <apiKey>  The API key for the symbol server.
  -d|--disable-buffering         Disable buffering when pushing to an HTTP(S) server to decrease memory usage.
  -n|--no-symbols                If a symbols package exists, it will not be pushed to a symbols server.
  --no-service-endpoint          Does not append "api/v2/package" to the source URL.
  --interactive                  Allow the command to block and require manual action for operations like authentication.
  --skip-duplicate               If a package and version already exists, skip it and continue with the next package in the push, if any.
```


```
❯ dotnet --version
7.0.202
```

```
❯ dotnet nuget --version
NuGet Command Line
6.5.0.136
```

**Comment by baronfel** [comment]
I think this was fixed in the 7.0.200 series, with this PR: https://github.com/dotnet/sdk/pull/29266

The SDK wasn't forwarding the correct set of arguments to NuGet for _NuGet_ to determine what kind of help to show.

**Comment by joelverhagen** [comment]
I can confirm this is fixed in 7.0.202. To reduce analysis work for the NuGet Client team, I will close this issue since it appears (with a high degree of confidence) that the issue is resolved.
